### PR TITLE
Nicholas Bestauros - Quiz 2

### PR DESCRIPTION
**Pull request by nic-best**
My methodology for testing was to retrieve all the data that the class will eventually print out, and test on that data. We were told to modify the signature to add return values. To me, that meant that I should return the data instead of using printstream as we did in homework 4.

To retrieve and test data:
To retrieve both the hashmap and the treemap data, I made an arraylist of type collection, and added both maps to the list.
I was then able to retrieve the entries from each of the lists and test from there. I made an auxillary testing method to ensure that this strategy of retrieving and testing the data was functioning. Then I made another test method that had ArrayLists of test strings, expectedWordMaps and expectedSortedMaps. I then iterated over the lists, adding each entry to a string that corresponds to each list, checking if each of the expected strings matched the string results of the data returned by wordCount.

Methodology behind creating test cases:
I brainstormed basic cases (such as one word) and grew from there. I also thought of the edge case of an empty string. The empty string actually presented a bug in the code (it would print =1 and 1=[]) when in reality, nothing should have been printed because an empty string doesnt have any words. The rest of my test cases all built upon the previous ones. I wanted to test the different behaviors of the class, such as the fact that the hashmap doesn't have a specific ordering (ordering reliant on the hashing function) while the treemap will have the natural ordering of the information given. In testing for these behaviors, each test also ensured that the function was counting all the words in the input String, and not adding any extra nor was the function miscounting any words or displaying the output in the incorrect order. This is due to the fact that I was comparing the given and the expected strings. The equals check on two strings will only be true if they are identical. As all the test cases are passing, the identical strings mean that the output is correct to the test cases.

### CODE REVIEW

File: src/test/java/HandleTruthTest.java [hunk 1]
```diff
@@ -1,8 +1,109 @@
...
+
+        //test numbers
+        testStrings.add("2 2 3 4 3 3 4 4 1 4");
+        expectedWordMaps.add("1=12=23=34=4");
+        expectedSortedMaps.add("4=[4]3=[3]2=[2]1=[1]");
+
+        for (int i = 0; i < testStrings.size(); i++) {
+            ArrayList<AbstractMap> mapList = HandleTruth.wordCount(testStrings.get(i));
```
Comment (pdgetrf) [inline src/test/java/HandleTruthTest.java:91]: here you can actually just return the sortedMap Map<Integer, Set<String>> and use the equals function for comparison.

Comment (pdgetrf) [reply src/test/java/HandleTruthTest.java:91]: Please see https://github.com/pdgetrf/you-cant-handle-the-truth/blob/solution/src/test/java/HandleTruthTest.java as an example.

**Review by pdgetrf — APPROVED** [review]
lgtm
